core/vm/gas.go

```diff
@@ -55,111 +70,37 @@ func callGas(gasTable params.GasTable, availableGas, base, callCost *big.Int) *b
 	}
 	return callCost
 }
+*/
 
-// baseCheck checks for any stack error underflows
-func baseCheck(op OpCode, stack *Stack, gas *big.Int) error {
-	// PUSH and DUP are a bit special. They all cost the same but we do want to have checking on stack push limit
-	// PUSH is also allowed to calculate the same price for all PUSHes
-	// DUP requirements are handled elsewhere (except for the stack limit check)
-	if op >= PUSH1 && op <= PUSH32 {
-		op = PUSH1
+// calcGas returns the actual gas cost of the call.
+//
+// The cost of gas was changed during the homestead price change HF. To allow for EIP150
+// to be implemented. The returned gas is gas - base * 63 / 64.
+func callGas(gasTable params.GasTable, availableGas, base uint64, callCost *big.Int) (uint64, error) {
+	if gasTable.CreateBySuicide > 0 {
+		availableGas = availableGas - base
+		gas := availableGas - availableGas/64
+		if callCost.BitLen() > 64 || gas < callCost.Uint64() {
+			return gas, nil
+		}
 	}
-	if op >= DUP1 && op <= DUP16 {
-		op = DUP1
+	if callCost.BitLen() > 64 {
+		return 0, errGasUintOverflow
 	}
 
-	if r, ok := _baseCheck[op]; ok {
-		err := stack.require(r.stackPop)
-		if err != nil {
-			return err
-		}
-
-		if r.stackPush > 0 && stack.len()-r.stackPop+r.stackPush > int(params.StackLimit.Int64()) {
-			return fmt.Errorf("stack limit reached %d (%d)", stack.len(), params.StackLimit.Int64())
-		}
+	return callCost.Uint64(), nil
+}
 
-		gas.Add(gas, r.gas)
-	}
-	return nil
+func toWordSize(size uint64) uint64 {
+	return (size + 31) / 32
```

Comment:
The same as above. The size would have to be very large for it to become an issue and there isn't enough ether for this. This number is always the results of the furthest memory byte, not the extra bytes added.